test/json_unittest.cc

```diff
@@ -156,11 +328,30 @@ TEST(TrivialToStringTest, ObjectEmpty) {
 TEST(TrivialParseTest, ObjectOneField) {
   GuardJsonException([](){
     json::value v = json::Parser::FromString("{\"f\":2}");
+    EXPECT_EQ(json::ObjectType, v->type());
+
+    EXPECT_FALSE(v->Is<json::Null>());
+    EXPECT_FALSE(v->Is<json::Boolean>());
+    EXPECT_FALSE(v->Is<json::Number>());
+    EXPECT_FALSE(v->Is<json::String>());
+    EXPECT_FALSE(v->Is<json::Array>());
+    EXPECT_TRUE(v->Is<json::Object>());
+
+    EXPECT_THROW(v->As<json::Null>(), json::Exception);
+    EXPECT_THROW(v->As<json::Boolean>(), json::Exception);
+    EXPECT_THROW(v->As<json::Number>(), json::Exception);
+    EXPECT_THROW(v->As<json::String>(), json::Exception);
+    EXPECT_THROW(v->As<json::Array>(), json::Exception);
     const auto& obj = v->As<json::Object>();
+
     EXPECT_EQ(1, obj->size());
-    EXPECT_TRUE(v->As<json::Object>()->Has("f"));
-    EXPECT_FALSE(v->As<json::Object>()->Has("g"));
+    EXPECT_TRUE(obj->Has("f"));
+    EXPECT_FALSE(obj->Has("g"));
```

Comment:
Let's group all `"f"`-related checks together (and all `"g"`-related checks to follow).

Reply:
Done.